core/game.js

```diff
@@ -85,32 +84,29 @@ export class Game {
   start() {
     this.running = true;
     this.last_tick = performance.now();
-    window.requestAnimationFrame((tick_time) => this.tick(tick_time));
+    window.requestAnimationFrame(frame_time => this.frame(frame_time));
   }
 
-  tick(tick_time) {
-    if (this.running) {
-      window.requestAnimationFrame((tick_time) => this.tick(tick_time));
+  emit(event_name, ...args) {
+    if (this.listeners.has(event_name)) {
+      for (const handler of this.listeners.get(event_name)) {
+        handler(...args);
+      }
     }
+  }
 
-    const next_tick = this.last_tick + this.tick_length;
-
-    if (tick_time > next_tick) {
-      const ticks_qty = Math.floor((tick_time - this.last_tick) / this.tick_length);
-      this.perform_ticks(ticks_qty);
-      this.render(this.last_tick);
+  add_listener(event_name, handler) {
```

Comment:
What do you think about names like `on` and `off`? It's a common pattern among js libraries and should save us few bytes :).

Reply:
Sure, I like it!  The shorter the names of the public API, the better!